softwareComponents/kinematics/include/shapes.hpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include <iostream>
+#include <variant>
+#include <vector>
+
+#include <configuration/Matrix.h>
+
+namespace rofi::geometry {
+
+using namespace rofi::configuration::matrices;
+
+struct Box {
```

Comment:
Nechceš zvážit mít box jako šablonovaný vectorem (něco jako `GenericBox`) a poskytovat `using Box = GenericBox< Vector >`? Občas se může hodit mít celočíselný strom. Stejně tak pro ostatní objekty.